sympy/core/tests/test_sympify.py

```diff
@@ -274,6 +275,11 @@ def test_lambda_raises():
 def test_sympify_raises():
     raises(SympifyError, lambda: sympify("fx)"))
 
+    class A:
+        def __str__(self):
+            return 'x'
+    raises(SymPyDeprecationWarning, lambda: sympify(A()))
```

Comment:
We should not use `raises` to test for warnings. We have `warns` and in particular `warns_deprecated_sympy`. Using `raises` fails under `pytest` and also means that you can't actually test the behaviour.

With warns you can do e.g.
```
    with warns_deprecated_sympy():
        assert sympify(x) == 1
```
and the assert will actually be tested as well as the fact that the warning was emitted.